### PR TITLE
Moving the PipeArray utility to the fpga_tools namespace

### PR DESCRIPTION
**Pull request by tyoungsc**
# Existing Sample Changes
## Description
Title says it all. Updated all samples using the shared utility.

## How Has This Been Tested?
- [X] Command Line

### CODE REVIEW

File: DirectProgramming/DPC++FPGA/include/pipe_utils.hpp [hunk 1]
```diff
@@ -121,12 +121,14 @@ struct write_currying<WriteFunc, BaseTy, std::index_sequence<I...>> {
...
 };
 
-}  // namespace
+}  // namespace detail
 
 // =============================================================
 // PipeArray
 // =============================================================
 
+namespace fpga_tools {
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/include/pipe_utils.hpp:130]: Should we put the detail namespace inside fpga_tools (i.e. move this to the top of the file)?  Not sure what the standard pattern is here.

**Review by mdbtucker — APPROVED** [review]
Approving to trigger CI

File: DirectProgramming/DPC++FPGA/include/pipe_utils.hpp (hunk 1 above)
Comment (tyoungsc) [reply DirectProgramming/DPC++FPGA/include/pipe_utils.hpp]: Yes. Done.

**Review by mdbtucker — APPROVED** [review]

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/src/gzipkernel_ll.cpp [hunk 2]
```diff
@@ -53,12 +53,13 @@
...
-using acc_dist_channel_array = PipeArray<  // Defined in "pipe_utils.hpp".
-    class dist_channel_pipe_id,            // An identifier for the pipe.
-    struct DistLen,                        // The type of data in the pipe.
-    32,                                     // The capacity of each pipe.
-    NUM_ENGINES                            // array dimension.
-    >;
+using acc_dist_channel_array =
+  fpga_tools::PipeArray<          // Defined in "pipe_utils.hpp".
+    class dist_channel_pipe_id,   // An identifier for the pipe.
+    struct DistLen,               // The type of data in the pipe.
```
Comment (keryell) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/src/gzipkernel_ll.cpp:59]: Curious mix of `class` and `struct` here.

Comment (tyoungsc) [reply DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/src/gzipkernel_ll.cpp:59]: The `struct` keyword is unnecessary here, but I think the use of a struct is fine: `DistLen` is just a bundle of a couple of types.

**Pull request merged**